src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -2136,4 +2141,104 @@ protected function appendOrderArgumentsToUrl($order_id, $order_key, $webhook_url
         );
         return $webhook_url;
 }
+
+    /**
+     * @param $new_order_status
+     *
+     * @param $statusTypeString
+     *
+     * @return mixed|void|null
+     */
+    protected function overwriteFilters($new_order_status, $statusTypeString)
+    {
+        // Overwrite plugin-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString,
+            $new_order_status
+        );
+
+        // Overwrite gateway-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString . $this->id,
+            $new_order_status
+        );
+        return $new_order_status;
+    }
```

Comment:
If this method is needed I would anyway convert it to 1 parameter method and move extra text from getting injected into the method. You can use the `$new_order_status` instead of passing two parameter one of which is just to produce an action name. 

Why not `Mollie_WC_Plugin::PLUGIN_ID . "_order_status_{$new_order_status}"`?

Indeed as it is now you have https://github.com/mollie/WooCommerce/pull/349/files#diff-d092edd23782109ce7e2e7e8b5279718R2215 which get passed an useless parameter because it's needed to creat the filter name.

Reply:
well I like the idea to make it simpler but 
const STATUS_ON_HOLD = 'on-hold';
$newOrderStatus = self::STATUS_ON_HOLD;
and the actions we need so we don't break anyone using from previous versions these filters are  '_order_status_on_hold' or '_order_status_refunded' I've added new '_order_status_partially_refunded' so this can be called whatever we want. Thoughts?

Reply:
@mmaymo What about my latest comment?